src/game.zig

```diff
@@ -378,13 +378,24 @@ pub const Player = struct { // MARK: Player
 		main.renderer.MeshSelection.breakBlock(&inventory__SEND_CHANGES_TO_SERVER.items[selectedSlot]);
 	}
 
-	pub fn acquireSelectedBlock() void { 
+	pub fn acquireSelectedBlock() void {
 		if (main.renderer.MeshSelection.selectedBlockPos) |selectedPos| {
 			const block = main.renderer.mesh_storage.getBlock(selectedPos[0], selectedPos[1], selectedPos[2]) orelse return;
 			for (0..items.itemListSize) |idx|{
 				if (items.itemList[idx].block == block.typ){
 					const item = items.Item {.baseItem = &items.itemList[idx]};
+					const slotItem = inventory__SEND_CHANGES_TO_SERVER.items[selectedSlot];
+					if (slotItem.empty() or !std.meta.eql(slotItem.item, item)) {
+						for (0..12) |slotIdx| {
+							if (inventory__SEND_CHANGES_TO_SERVER.items[slotIdx].empty()) {
+								inventory__SEND_CHANGES_TO_SERVER.items[slotIdx] = items.ItemStack {.item = item, .amount = items.itemList[idx].stackSize};
+								selectedSlot = @intCast(slotIdx);
+								return;
+							}
+						}
+					}
 					inventory__SEND_CHANGES_TO_SERVER.items[selectedSlot] = items.ItemStack {.item = item, .amount = items.itemList[idx].stackSize};
+					return;
```

Comment:
Also, please change this to a break, a return might have unintended consequences in the future.